File: JAVA/FUNCIONAL/superior.java
public class superior {                    // Clase principal
    
    static void ejecutar(Runnable accion) {     // Función que recibe otra acción
        accion.run();                           // Ejecutar la acción recibida
    }                                           // Runnable es tipo para funciones
    
    public static void main(String[] args) {    // Método principal
        ejecutar(() -> System.out.println("Hola")); // Pasar función directa (lambda)
    }
}

File: PYTHONGUIA/FUNCIONAL/superior.py
def saludar():                                  # Función normal (la acción)
    print("Hola")                               # Imprimir saludo

def ejecutar(func):                             # Función que recibe otra función
    func()                                      # Ejecutar la función recibida

ejecutar(saludar)                               # Pasar 'saludar' como dato
